internal/service/advancedcluster/model_advanced_cluster.go

```diff
@@ -942,8 +956,11 @@ func expandRegionConfigSpec(tfList []any, providerName string, rootDiskSizeGB *f
 		apiObject.NodeCount = conversion.Pointer(v.(int))
 	}
 
-	// TODO: CLOUDP-258270 once disk_size_gb schema attribute is added at this level, we will check if defined in config and prioritize over value defined at root level (deprecated and to be removed)
 	apiObject.DiskSizeGB = rootDiskSizeGB
+	// disk size gb defined in inner level will take precedence over root level.
+	if v, ok := tfMap["disk_size_gb"]; ok && v.(float64) != 0 {
```

Comment:
it's a best practice to make sure conversion can be done, e.g. using v.(float64) with ok param

Reply:
Synced offline. For resource using sdk v2 this is a common pattern as the schema is already ensuring a certain type for each attribute meaning this can be considered a safe cast. 
Main con of handling the error is that is would add significant boilerplate for each time we access a value from the TF model. This was solved in the terraform plugin framework as we parse the TF model into a typed struct.
cc: @EspenAlbert as you were seeing these warning in the new linter.